mds/apis/prv_api/service_areas.py

```diff
@@ -81,19 +98,74 @@ def create(self, *args, **kwargs):
     def update(self, *args, **kwargs):
         return super()._update(*args, **kwargs)
 
+    @action(methods=["post"], url_path="import", detail=False)
+    def import_polygons(self, request, pk=None):
+        polygons = request.data.get("polygons", None)
+
+        if not isinstance(polygons, list):
+            return Response(status=400)
+
+        try:
+            polygons_to_create = []
+            for polygon in polygons:
+                geom = polygon.get("geom", None)
+                if geom and geom["type"] == "Polygon":
+
+                    areas = []
+                    for area_label in polygon.get("areas", []):
+                        defaults = {"color": "#%06x" % random.randint(0, 0xFFFFFF)}
+                        # Create new Area if doesn't exist (based on label)
+                        area = models.Area.objects.get_or_create(
+                            label=area_label, defaults=defaults
+                        )[0]
+                        areas.append(area)
+                    poly = models.Polygon(
+                        label=polygon.get("label", ""), geom=str(geom)
+                    )
+                    poly.areas.set([a.id for a in areas])
+                    polygons_to_create.append(poly)
+            models.Polygon.objects.bulk_create(polygons_to_create)
+        except IntegrityError as ex:
+            return Response(exception=ex, status=500)
```

Comment:
?? You should not return 500 errors to your users. Either a 400 if there is something wrong with their data in this case you should tell them what is wrong.